spec/date.html

```diff
@@ -708,76 +712,46 @@ <h1>DifferenceDate ( _y1_, _m1_, _d1_, _y2_, _m2_, _d2_, _largestUnit_ )</h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-temporal-totemporaldaterecord" aoid="ToTemporalDateRecord">
-      <h1>ToTemporalDateRecord ( _temporalDateLike_ )</h1>
+    <emu-clause id="sec-temporal-totemporaldatefields" aoid="ToTemporalDateFields">
+      <h1>ToTemporalDateFields ( _temporalDateLike_, _calendar_ )</h1>
       <emu-alg>
         1. Assert: Type(_temporalDateLike_) is Object.
-        1. If _temporalDateLike_ has an [[InitializedTemporalDate]] internal slot, then
-          1. Return the Record {
-              [[Year]]: _temporalDateLike_.[[Year]],
-              [[Month]]: _temporalDateLike_.[[Month]],
-              [[Day]]: _temporalDateLike_.[[Day]],
-            }.
-        1. Let _result_ be a new Record with all the internal slots given in the Internal Slot column in <emu-xref href="#table-temporal-temporaldatelike-properties"></emu-xref>.
-        1. For each row of <emu-xref href="#table-temporal-temporaldatelike-properties"></emu-xref>, except the header row, in table order, do
-          1. Let _property_ be the Property value of the current row.
+        1. Let _fieldsMethod_ be ? Get(_calendar_, *fields*).
+        1. Let _fields_ be ? CreateArrayFromList(« *"day"*, *"era"*, *"month"*, *"year"* »).
+        1. Set _fields_ to ? Call(_fieldsMethod_, _calendar_, « _fields_ »).
+        1. Let _result_ be ? OrdinaryObjectCreate(%Object.prototype%).
+        1. For each value _property_ of _fields_, do
           1. Let _value_ be ? Get(_temporalDateLike_, _property_).
-          1. If _value_ is *undefined*, then
+          1. If _property_ is one of *"day"*, *"month"*, or *"year"*, and _value_ is *undefined*, then
             1. Throw a *TypeError* exception.
-          1. Let _value_ be ? ToInteger(_value_).
-          1. Set _result_'s internal slot whose name is the Internal Slot value of the current row to _value_.
+          1. If _property_ is not *"era"*, then
+            1. Let _value_ be ? ToInteger(_value_).
```

Comment:
This makes no sense to me: this would convert any custom property to an integer?

Reply:
That is what I intended. The alternative is to treat any custom property just like `era` and leave it unchanged. My reasoning was that we wouldn't want to encourage more non-numerical properties like era, but I don't have a strong opinion on which is correct.

Reply:
We shouldn't assert a particular type on custom properties, except *perhaps* requiring that it is a primitive.

Reply:
I don't have a clear idea of the use case of this feature, but this seems to make it impossible to use non-integer properties, rather than just discourage. I'll merge now in the interest of moving forward, but filed #1045.